code/Modules/Props/Types/Machine.cs

```diff
@@ -0,0 +1,53 @@
+﻿using ssl.Modules.Props.Data;
+
+namespace ssl.Modules.Props.Types
+{
+    /// <summary>
+    /// A machine is a basic classs for all machines in the map. A machine have a health and it is reparaible by the mecanic.
+    /// It's not instanciable because a machine have to had a purpose.
+    /// </summary>
+    public abstract class Machine : Prop<MachineData>
+    {
+        /// <summary>
+        /// Health of a basic machine
+        /// </summary>
+        private int health = 10;
```

Comment:
Use the Health property already on the entity.
It may even be overriden in the Prop class alongside the OnTakeDamage etc methods